hack/release/common.sh

```diff
@@ -110,6 +104,41 @@ publishHelmChart() {
   cosignOciArtifact "${version}" "${commit_sha}" "${build_date}" "${oci_repo}${helm_chart}:${version}@${helm_chart_digest}"
 }
 
+updateAhConfig() {
+  local oci_repo helm_chart ah_config_path image_config_path image_config media_type oci_repository oci_image old_config_digest blob_digest
+
+  oci_repo="${1}"
+  helm_chart="${2}"
+
+  ah_config_path="./charts/${helm_chart}/artifacthub-repo.yml"
+
+  if [[ -f "${ah_config_path}" ]] && [[ "${oci_repo}" == "${RELEASE_REPO_ECR}" ]]; then
+    # ECR requires us to create an empty config file for an alternative
+    # media type artifact push rather than /dev/null
+    # https://github.com/aws/containers-roadmap/issues/1074
+    image_config_path="$(mktemp)"
+    echo "{}" > "${image_config_path}"
+
+    image_config="${image_config_path}:application/vnd.cncf.artifacthub.config.v1+yaml"
+    media_type="application/vnd.cncf.artifacthub.repository-metadata.layer.v1.yaml"
+    oci_repository="${oci_repo}${helm_chart}"
+    oci_image="${oci_repository}:artifacthub.io"
+
+    old_config_digest="$(crane digest "${oci_image}" || true)"
+
+    if [[ -n "${old_config_digest}" ]]; then
```

Comment:
Out of curiosity: What's the technical problem if each new release produces a new digest for the ArtifactHub config file?

Reply:
The config file digest stays the same, but the manifest file has a build date which obviously changes every time (I can't find a reproducible build flag although explicitly setting the annotation might work). So the outcome is an orphaned image every time the chart releases, as the `artifacthub.io` tag is moved.